Basic error handling for usage reports

marcospri opened the pull request:
Give some indication that the error presented to the user is caused by the organization not having any activity.


### Testing 

- Try to generate a report over http://localhost:8001/admin/org/1/usage with dates with no local activity.

robertknight commented on tests/unit/lms/services/organization_service_test.py:
```diff
@@ -279,6 +279,24 @@ def test_usage_report(self, svc, org_with_parent, h_api):
+        since = datetime(2023, 1, 1)
+        until = datetime(2023, 12, 31)
+
+        with pytest.raises(ValueError):
```
Since `ValueError` is a very general exception, it would be good to assert on relevant contents of the message. Same for the other tests. Otherwise the test could still pass despite the wrong error being shown to the user.

marcospri replied:
Coverage would assure that overall all the exception are tested but yeah this test will individually be testing the wrong thing.

Added some light asserting of the message value :+1:

marcospri commented:
Added a commit to rename the test file which was not matching the original source file:

```
renamed:    tests/unit/lms/services/organization_service_test.py -> tests/unit/lms/services/organization_test.py
```

robertknight reviewed (commented):
This works and it presents an error message, but it seems weird to return a 4xx error page rather than an empty result set. In the context of a search service I would expect a syntax error in the query and an empty result set to be handled differently.

The error response I see:

<img width="849" alt="Error response" src="https://github.com/hypothesis/lms/assets/2458/bf429e49-5168-402b-875f-0540efe5e7d2">

marcospri commented:
> In the context of a search service I would expect a syntax error in the query and an empty result set to be handled differently.

Not sure what you mean, while generating a report returning an empty set will not give any context to the user, here although if it's a bit rough you get some indication of why you don't get anything (no activity)

robertknight commented:
> Not sure what you mean, while generating a report returning an empty set will not give any context to the user,

My point was that the error message here makes it look like something went wrong with the service, or there was something wrong with the request, since it is what you see for eg. 5xx responses. However an empty response is not an error with the request, or an unexpected backend failure, it's just a search that returned no results.

Can I confirm I'm understanding the goal here. My understanding is that we're trying to provide the user with an explanation of why this table is empty:

<img width="1375" alt="Usage report" src="https://github.com/hypothesis/lms/assets/2458/0253cb77-8f18-41c6-8a63-0927758046fb">

Have I got this right? What I'd expect to see is that the form looks like the above, but with the part in red replaced by a message that tells the user why the result is empty.

marcospri commented:
@robertknight Sorry for the very(very) slow turnaround on this one.

We don't have that many mechanisms in the admin pages to make this ideal but I reckon the last solution here is acceptable, we flash a message and return and empty report, it looks like:


![Screenshot from 2024-02-07 14-55-28](https://github.com/hypothesis/lms/assets/1433832/ed288b62-a960-4dfc-a50e-c227af9ec2ab)

robertknight reviewed (approved):
OK, this sounds better. In polished end-user facing UIs I would present cases where the input is valid, but there were simply no results, with a special "empty results" view in the table.
